File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {

        HashMap<Character,Integer>x=new HashMap<>();
boolean check=false;
        for(int i=0;i<s.length();i++){
            char ch =s.charAt(i);
            x.put(ch,x.getOrDefault(ch,0)+1);

        }
        int sum=0;
       for( Map.Entry<Character,Integer> p:x.entrySet()){
        if((p.getValue())%2==0){
            sum+=p.getValue();
        }else {
            sum+=p.getValue()-1;
            check=true;
        }
       }
       if(check){
          return sum+1;
       }
        return sum;
    }
}

File: 0409-longest-palindrome/0409-longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char,int>ans;

        for(int i=0;i<s.size();i++){
            ans[s[i]]++;
        }
        int sum=1;
        bool  p=false;
        for(auto &x:ans){
            if(x.second>=2){
                int a=x.second/2;
                int b=a*2;
                sum+=b;
                if(b%2!=0){
                    p=true;
                }
            }
            
        }
        if(p==true){
            sum+=1;
        }
        return sum;
    }
};